doc/rst/language/spec/types.rst

```diff
@@ -589,3 +589,202 @@ generic (:ref:`Type_Aliases_in_Generic_Types`).
 
       2
 
+
+.. _Querying_the_Type_of_an_Expression:
+
+Querying the Type of an Expression
+----------------------------------
+
+.. code-block:: syntax
+
+   type-query-expression:
+     expression . `type'
+
+The type of a an expression can be queried with ``.type``. This
+functionality is particularly useful when doing generic programming
+(see :ref:`Chapter-Generics`).
+
+   *Example (dot-type.chpl)*.
+
+   For example, this code uses ``.type`` to query the type of the
+   variable ``x`` and store that in the type alias ``t``:
+
+   .. code-block:: chapel
+
+      var x: int;
+      type t = x.type;
+
+   .. BLOCK-test-chapelpost
+
+      writeln(t:string);
+
+   .. BLOCK-test-chapeloutput
+
+      int(64)
+
+   *Open issue*.
+
+   Given a nested expression that has ``.type`` called on it,
+   for example ``f()`` in ``f().type``, in which circumstances should
+   ``f()`` be evaluated for side effects?
+
+   At first it might seem that ``f()`` should never be evaluated for side
+   effects. However, it must be evaluated for side effects if ``f()`` an
+   array or domain type, as these have a runtime component (see
+   :ref:`Types_with_Runtime_Components`). As a result, should ``f()`` in
+   such a setting always be evaluated for side effects?  The answer to
+   this question also also connected to the question of whether or not a
+   when a function returning a ``type`` is evaluated for side effects at
+   runtime.
+
+   One approach might be to introduce different means to query only the
+   compile-time component of the type or only the runtime component of
+   the time.
+
+
+.. _Operations_Available_on_Types:
+
+Operations Available on Types
+-----------------------------
+
+This section discusses how type expressions can be used. Type expressions
+include types, type aliases, ``.type`` queries, and calls to functions
+that use the ``type`` return intent.
+
+A type expression can be used to indicate the type of a value, as with
+``var x: typeExpression;`` (see :ref:`Variable_Declarations`).
+
+A type expression can be passed to a ``type`` formal of a generic
+function (see :ref:`Formal_Type_Arguments`).
+
+See also the :mod:`Types` module documentation which provides many
+functions to query properties of types.
```

Comment:
"The :mod:\`Types\` module provides many functions to query properties of types."